sunpy/visualization/wcsaxes_compat.py

```diff
@@ -144,23 +155,28 @@ def wcsaxes_heliographic_overlay(axes, grid_spacing: u.deg = 10*u.deg, annotate=
     else:
         raise ValueError("grid_spacing must be a Quantity of length one or two.")
 
+    if system == 'stonyhurst':
+        overlay = axes.get_coords_overlay(HeliographicStonyhurst(obstime=obstime, rsun=rsun))
+    elif system == 'carrington':
+        overlay = axes.get_coords_overlay(HeliographicCarrington(observer=observer, rsun=rsun))
```

Comment:
Hmm...  I think you should create the frame with an `obstime` if it is provided.  `observer.obstime` could conceivably be different from the desired `obstime` of the Carrington grid